src/core/service.c

```diff
@@ -2758,6 +2760,102 @@ static void service_run_next_main(Service *s) {
         (void) service_set_main_pidref(s, TAKE_PIDREF(pidref), &s->main_command->exec_status.start_timestamp);
 }
 
+static int extension_name_has_vpick(char *path) {
+        _cleanup_free_ char *dir = NULL;
+        int r;
+
+        assert(path);
+
+        r = path_extract_filename(path, &dir);
+        if (r == -EADDRNOTAVAIL)
+                return 0;
+        if (r < 0)
+                return r;
+
+        return endswith(dir, ".v") ? 1 : 0;
+}
+
+static int service_has_vpick_extensions(Service *s) {
+        int r;
+
+        assert(s);
+
+        FOREACH_ARRAY(mi, s->exec_context.extension_images, s->exec_context.n_extension_images) {
+                r = extension_name_has_vpick(mi->source);
+                if (r < 0)
+                        return r;
+                if (r > 0)
+                        return 1;
+        }
+
+        STRV_FOREACH(ed, s->exec_context.extension_directories) {
+                r = extension_name_has_vpick(*ed);
+                if (r < 0)
+                        return r;
+                if (r > 0)
+                        return 1;
+        }
```

Comment:
These really should reside in `execute.c`, given that you're operating on `ExecContext` rather than `Service`.

Reply:
This makes sense! Goes well with the `exec_context_has_..` series of functions.